plugins/nisq_analyzer.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2021 QHAna plugin runner contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from http import HTTPStatus
+from typing import Optional
+from os import environ
+
+from flask import abort
+from flask.app import Flask
+from flask.helpers import url_for
+from flask.views import MethodView
+from marshmallow import EXCLUDE
+
+from qhana_plugin_runner.api.plugin_schemas import (
+    EntryPoint,
+    PluginMetadata,
+    PluginMetadataSchema,
+    PluginType,
+)
+from qhana_plugin_runner.api.util import (
+    FrontendFormBaseSchema,
+    SecurityBlueprint,
+)
+from qhana_plugin_runner.util.plugins import QHAnaPluginBase, plugin_identifier
+
+_plugin_name = "nisq-analyzer"
+__version__ = "v0.1.0"
+_identifier = plugin_identifier(_plugin_name, __version__)
+
+
+NISQ_BLP = SecurityBlueprint(
+    _identifier,  # blueprint name
+    __name__,  # module import name!
+    description="NISQ Analyzer Plugin.",
+)
+
+
+@NISQ_BLP.route("/")
+class PluginsView(MethodView):
+    """Plugins collection resource."""
+
+    @NISQ_BLP.response(HTTPStatus.OK, PluginMetadataSchema())
+    @NISQ_BLP.require_jwt("jwt", optional=True)
+    def get(self):
+        """Endpoint returning the plugin metadata."""
+        plugin = NisqAnalyzer.instance
+        if plugin is None:
+            abort(HTTPStatus.INTERNAL_SERVER_ERROR)
+        return PluginMetadata(
+            title=plugin.name,
+            description=plugin.description,
+            name=plugin.name,
+            version=plugin.version,
+            type=PluginType.processing,
+            entry_point=EntryPoint(
+                href=url_for(f"{NISQ_BLP.name}.PluginsView"),
+                ui_href=f"http://localhost:{plugin.port}",
```

Comment:
Make the whole of the nisq analyzer URL configurable! Like this it must run on localhost.